source/agora/common/Block.d

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+
+    Defines the data structure of a block
+
+    The design is influenced by Bitcoin, but will be ammended later.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.Block;
+
+import agora.common.Data;
+import agora.common.Transaction;
+
+
+/*******************************************************************************
+
+    The block header which contains a link to the previous block header,
+    unless it's the genesis header.
+
+*******************************************************************************/
+
+public struct BlockHeader
+{
+    /// Hash of the previous block in the chain of blocks
+    public Hash prev_block;
+
+    /// Block height (genesis is #1)
```

Comment:
Why `#1` ? :P

Reply:
Because it's the best block ever. Honestly though, maybe as a safety check? if block height = 0 (default initializer), something went wrong. I don't really mind one way or the other.